### DISCUSSION TITLE
How to keep the planet_osm_ways table without using the --slim option?

### DISCUSSION THREAD

**Discussion by afarber** [osm2pgsql-dev/osm2pgsql#2154] [Q&A] [answered]
Hello dear osm2pgsql developers and users, when I run the following command:

`osm2pgsql --username=$PGUSER --database=$PGDATABASE --create --cache=60000 --hstore --latlong /data/map.osm.pbf`

then the **planet_osm_ways** table is not generated and there are only these tables:

```
osm_database=> \dt
               List of relations
 Schema |        Name        | Type  |  Owner   
--------+--------------------+-------+----------
 public | planet_osm_line    | table | osm_user
 public | planet_osm_point   | table | osm_user
 public | planet_osm_polygon | table | osm_user
 public | planet_osm_roads   | table | osm_user
 public | spatial_ref_sys    | table | postgres
(5 rows)
```

I do need that table though, because my application gets only OSM node ids from OSRM and then needs to resolve them to parent OSM way ids.

Adding the `-S default.style` as in the command below has not helped:

`osm2pgsql --username=$PGUSER --database=$PGDATABASE --create -S /usr/share/osm2pgsql/default.style --cache=60000 --hstore --latlong /data/map.osm.pbf`

I do not want to introduce `--slim` option, because with it my Europe OSM import takes 10 hours instead of the current 3 hours.

Thank you for any ideas
Alex

**Comment by joto** [comment, marked as answer]
You have to use `--slim` to get the `planet_osm_ways` table. It takes so long because that table is huge. No way around that.

**Reply by afarber** [reply]
Thank you for the quick response @joto !

If I reintroduce the `--slim`, should I drop the `--cache=60000` option? 

Or is the latter option still useful?

**Reply by joto** [reply]
The `--cache` option actually only does something in slim mode, so yes, this might help, depending on how much memory you have etc. You should also look into the `--flat-nodes` option. See the [manual](https://osm2pgsql.org/doc/manual.html) for more information.
